src/test/parse.ts

```diff
@@ -114,6 +118,25 @@ export function parseErrorsFromMarkup(text: string): LintError[] {
     return lintErrors;
 }
 
+function substituteMessage(substitutions: Map<string, string>, message: string): string {
+    const substitution = substitutions.get(message);
+    if (substitution !== undefined) {
+        return substitution;
+    }
+    return formatMessage(substitutions, message);
+}
+
+function formatMessage(substitutions: Map<string, string>, message: string): string {
+    const formatMatch = /^([\w_]+) % \(\s*([^,]+(?:\s*,\s*[^,]+)*)\s*\)$/.exec(message);
```

Comment:
add a comment somewhere about the expected format of `message` and the substitutions since the regex is hard to parse